src/charm.py

```diff
@@ -62,11 +78,11 @@ def container(self) -> Container:
             The controller of the flask application workload container.
 
         Raises:
-            RuntimeError: if the pebble service inside the container is not ready while the
+            PebbleNotReadyError: if the pebble service inside the container is not ready while the
                 ``require_connected`` is set to True.
         """
         if not self.container_can_connect():
-            raise RuntimeError("pebble inside flask-app container is not ready")
+            raise PebbleNotReadyError("pebble inside flask-app container is not ready")
```

Comment:
Why not logging and deferring the event here instead of throwing an exception?

Reply:
This method can be used in different context, and not all context have the access to the event object.
Also, charm should stop executing any code after the event deferring, which is not possible to achieve inside the `container` method.